Add note about class reading in class loading guide (#7991)
quarkusio/quarkus: docs/src/main/asciidoc/class-loading-reference.adoc

For all other use cases (e.g. tests, dev mode, and building the application) Quarkus
uses the class loading architecture outlined here.

== Reading Class Bytecode

Whenever class bytecode needs to be read, a `@BuildStep` where class reading is done have
to have `loadsApplicationClasses` parameter set to `true`. Otherwise the class will not be

> found.

Reviewer: ```suggestion
Whenever class bytecode needs to be read, a `@BuildStep` where class reading is done must have the `loadsApplicationClasses` parameter set to `true`. Otherwise the class will not be found.
```

> It's also important to use correct `ClassLoader`. Best is to get it by calling the

Reviewer: ```suggestion
It's also important to use the correct `ClassLoader`. The recommended approach is to get it by calling the
```

`Thread.currentThread().getContextClassLoader()` method.

> Example `@BuildStep` may be similar to the following:

Reviewer: ```suggestion
Example:
```
